core/lib/progressbar.spec.ts

```diff
@@ -0,0 +1,57 @@
+import {beforeEach, describe, expect, test} from 'vitest';
+import {createProgressbar, getProgressbarDefaultConfig} from './progressbar';
+import type {ProgressbarWidget} from './progressbar';
+import type {WidgetState} from './types';
+
+describe(`Progressbar`, () => {
+	let progressbar: ProgressbarWidget;
+	let state: WidgetState<ProgressbarWidget>;
+
+	beforeEach(() => {
+		progressbar = createProgressbar();
+		progressbar.state$.subscribe((newState) => {
+			state = newState;
+		});
+	});
+
+	test(`should create progressbar with a default state`, () => {
+		expect(state).toEqual({
+			animated: false,
+			ariaLabel: 'Progressbar',
+			className: '',
+			finished: false,
+			height: '',
+			max: 100,
+			min: 0,
+			percentage: 0,
+			showValue: false,
+			slotContent: undefined,
+			slotDefault: undefined,
+			started: false,
+			striped: false,
+			value: 0,
+		});
+	});
+
+	test(`should calculate the percentage properly`, () => {
+		progressbar.patch({min: 1, max: 5, value: 4});
```

Comment:
In your test, you consider only a part of the state, meaning that a lot of use case may not be tested. For example, in this example (and in the UT of this file), the property `started` is not tested, but yet it changes from `true`to `false`.

Why not testing the full state directly ? 

For example

```
```suggestion
		let expectedState = state;
		progressbar.patch({min: 1, max: 5, value: 4});
		expectedState = {...expectedState, min: 1, max: 5, percentage: 75, value: 4, started: true};
		expect(state).toStrictEqual(expectedState);
		progressbar.patch({min: 10, max: 0});
		expectedState = {...expectedState, percentage: 0, min: 10, max: 10, value: 10, started: false, finished: true};
		expect(state).toStrictEqual(expectedState);
```

So that you'll be sure of all side effect of your api.